### PR TITLE
Remove stdlib and runtime dependencies on Foundation and CF

### PR DESCRIPTION
**Pull request by Catfish-Man**
Relanding https://github.com/apple/swift/pull/26630, with some fixes and improvements (now also decouples from CF properly!)

Fixes rdar://34198279

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test osx platform

**Comment by Catfish-Man** [comment]
@swift-ci please test os x platform

**Review by stephentyrone — APPROVED** [review]

File: stdlib/public/core/ContiguousArrayBuffer.swift [hunk 1]
```diff
@@ -417,13 +417,27 @@ internal struct _ContiguousArrayBuffer<Element>: _ArrayBufferProtocol {
...
 
 #if _runtime(_ObjC)
+  
   /// Convert to an NSArray.
   ///
   /// - Precondition: `Element` is bridged to Objective-C.
   ///
   /// - Complexity: O(1).
-  @inlinable
+  @usableFromInline
```
Comment (milseman) [inline stdlib/public/core/ContiguousArrayBuffer.swift]: I know these seems pedantic, but could you add a comment after the `@usableFromInline`? Something to the effect of:

```swift
@usableFromInline // was @inlinable through Swift 5.1 (see comment below)
```

I want some indication right on the decl of the binary compatibility story, without reading comments inside the body (which may be about the implementation).

**Comment by milseman** [comment]
Congrats!

File: stdlib/public/core/ContiguousArrayBuffer.swift (hunk 1 above)
Comment (stephentyrone) [reply stdlib/public/core/ContiguousArrayBuffer.swift]: Can we also mark these deprecated at this point?

Comment (Catfish-Man) [reply stdlib/public/core/ContiguousArrayBuffer.swift]: Can we mark things as deprecated without breaking our own/Foundation's callsites?

**Pull request merged**